aws/resource_aws_vpn_connection.go

```diff
@@ -618,55 +617,29 @@ func xmlConfigToTunnelInfo(xmlConfig string) (*TunnelInfo, error) {
 	return &tunnelInfo, nil
 }
 
-func validateVpnConnectionTunnelPreSharedKey(v interface{}, k string) (ws []string, errors []error) {
-	value := v.(string)
-
-	if (len(value) < 8) || (len(value) > 64) {
-		errors = append(errors, fmt.Errorf("%q must be between 8 and 64 characters in length", k))
-	}
-
-	if strings.HasPrefix(value, "0") {
-		errors = append(errors, fmt.Errorf("%q cannot start with zero character", k))
-	}
-
-	if !regexp.MustCompile(`^[0-9a-zA-Z_.]+$`).MatchString(value) {
-		errors = append(errors, fmt.Errorf("%q can only contain alphanumeric, period and underscore characters", k))
-	}
-
-	return
+func validateVpnConnectionTunnelPreSharedKey() schema.SchemaValidateFunc {
+	return validation.All(
+		validation.StringLenBetween(8, 64),
+		validation.StringDoesNotMatch(regexp.MustCompile(`(?i)^0`), "cannot start with zero character"),
```

Comment:
@megubyte just curious if the case-insensitive mode param here is necessary since we're scanning for the 0 num char

Reply:
Hello @anGie44! 👋 Apologies about this - it seems this has probably been a bit of a copy-paste issue on my part - I've been working on some other files in relation to the original ticket, and have simply just not noticed the case-insensitive mode param was there when pasting it.

I can remove this before it gets merged if you'd like. 😄

Reply:
no worries! i believe in this case we can omit it here to maintain readability but if there is an edge case you might be considering, we can definitely leave as is :)